docs/guide/CSS.md

```diff
@@ -0,0 +1,395 @@
+
+Class name selectors may be _chained_ together by appending another full stop and class name. The selector will match a widget that has _all_ of the class names set. For instance, the following sets a red background on widgets that have both `error` _and_ `disabled` class names.
+
+```sass
+.error.disabled {
+  background: darkred;
+}
+```
+
+Unlike the `id` attribute a Widget's classes can be changed after the Widget was created. Adding and removing CSS classes is the recommended way of changing the display while your app is running. There are a few methods you can use to manage CSS classes.
+
+- [add_class()][textual.dom.DOMNode.add_class] Adds one or more classes to a widget.
+- [remove_class()][textual.dom.DOMNode.remove_class] Removes class name(s) from a widget.
+- [toggle_class()][textual.dom.DOMNode.toggle_class] Removes a class name if it is present, or adds the name if its not already present.
+- [has_class()][textual.dom.DOMNode.has_class] Checks if a class(es) is set on a widget.
+- [classes][textual.dom.DOMNode.classes] Is a frozen set of the class(es) set on a widget.
+
+
+### Universal selector
+
+The _universal_ selectors is denoted by an asterisk and will match _all_ widgets.
+
+For example, the following will draw a red outline around all widgets:
+
+```sass
+* {
+  outline: solid red;
+}
+```
+
+### Pseudo classes
+
+Pseudo classes can be used to match widgets in a particular state. Psuedo classes are set automatically by Textual. For instance, you might want a button to have a green background when the mouse cursor moves over it. We can do this with the `:hover` pseudo selector.
+
+```sass
+Button:hover {
+  background: green;
+}
+```
+
+The `background: green` is only applied to the Button underneath the mouse cursor. When you move the cursor away from the button it will return to its previous background color.
+
+Here are some other such pseudo classes:
+
+- `:focus` Matches widgets which have input focus.
+- `:focus-within` Matches widgets with a focused a child widget.
+
+## Combinators
+
+More sophisticated selectors can be created by combining simple selectors. The rule that combines selectors is know as a _combinator_.
+
+### Descendant combinator
+
+If you separate two selectors with a space it will match widgets with the second selector that have a parent that matches the first selector.
+
+Here's a section of DOM to illustrate this combinator:
+
+<div class="excalidraw">
+--8<-- "docs/images/descendant_combinator.excalidraw.svg"
+</div>
+
+Let's say we want to make the text of the buttons in the dialog bold, but we _don't_ want to change the Button in the sidebar. We can do this with the following rule:
+
+```sass hl_lines="1"
+#dialog Button {
+  text-style: bold;
+}
+```
+
+The `#dialog Button` selector matches all buttons that are below the widget with an id of "dialog". No other buttons will be matched.
```

Comment:
This matches `Button` instances directly below `#dialog` or at any level of nesting under `#dialog`?